fastcrypto-tbls/src/random_oracle.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use digest::Digest;
+use serde::Serialize;
+use sha3::Sha3_512;
+
+/// Random Oracle from Sha3.
+/// - prefix should be globally unique.
+/// - evaluate serializes the given input and outputs SHA3-512(prefix_len as u32 | prefix | input).
```

Comment:
why 512?

Reply:
Given a uniform string x of length l, x mod q has a bias of 2^-(l-|q|), and so assuming we want <2^-128 bias we need l 256+128 for all the groups we work with. 512 seems like the right size.